packages/odk/src/Utils.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * If you have any helper functions which are NOT operations,
+ * you should add them here
+ */
+
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  logResponse,
+} from '@openfn/language-common/util';
+
+export const authorize = state => {
+  const auth = state.configuration;
+  if (auth.access_token) {
+    return state;
+  }
+
+  const email = auth.username;
+  const password = auth.password;
+
+  if (email && password) {
+    const options = {
+      body: JSON.stringify({ email, password }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+      baseUrl: auth.baseUrl,
+      parseAs: 'json',
+      maxRedirections: 1,
+    };
+    return commonRequest('POST', '/v1/sessions', options).then(response => {
+      return {
+        ...state,
+        configuration: {
+          ...state.configuration,
+          access_token: response.body.token,
+        },
+      };
+    });
+  }
+};
+export const prepareNextState = (state, response, callback = s => s) => {
+  const { body, ...responseWithoutBody } = response;
+
+  if (!state.references) {
+    state.references = [];
+  }
+
+  const nextState = {
+    ...composeNextState(state, response.body),
+    response: responseWithoutBody,
+  };
+
+  return callback(nextState);
+};
+
+function removeLeadingSlash(url) {
```

Comment:
I've love to find a better solution to this. It's tracked in https://github.com/OpenFn/adaptors/issues/578

This is fine as it for now - I just want to link the issue to this PR.